mbed_targets/config.py

```diff
@@ -17,7 +17,7 @@
 import os
 import dotenv
 
-dotenv.load_dotenv(dotenv.find_dotenv(usecwd=True), override=True)
+dotenv.load_dotenv(dotenv.find_dotenv(usecwd=True))
```

Comment:
Don't we need to actually read from the env at the call site? We cache the variable at import time so we could get out of sync with the user's shell environment?

Reply:
Isn't this true of this and previous approach where variables were assigned at load time? This has not been introduced by this PR, right?

Reply:
It wasn't true in the original code (i.e when this lived in `target_database`) as we read from the env in `_get_request`. Must've changed when it was moved here, and I missed it in the review.

Reply:
Could we address that separately then? I'd be keen not to pollute this PR with unrelated bugfix. :)

Reply:
Yeah, makes sense.